crates/router/src/core/payment_methods/cards.rs

```diff
@@ -45,7 +51,26 @@ pub async fn create_payment_method(
     customer_id: &str,
     payment_method_id: &str,
     merchant_id: &str,
+    connector: Option<&api::ConnectorData>,
```

Comment:
whenever token is there then connector also should be present right, create a tuple or struct and use the struct/tuple as `Option`

Reply:
I have passed connector and token in the next functions, token is not passed  as it is not necessary at this place.